### PR TITLE
Update Bazel to 6.4.0

### PR DESCRIPTION
**Pull request by avdv**
- update nixpkgs to nixos-23.11 which was released yesterday
- fix links in README and docs
- note that Bazel >= 6.0 is required in the README

### CODE REVIEW

**Comment by avdv** [comment]
macos CI jobs are now failing with:
```
ar: @../../../bazel-out/darwin-fastbuild/bin/external/_main~stack_snapshot~stackage/primitive-0.7.3.0/primitive-0.7.3.0/build/objs-6725/ar6725-2.rsp: No such file or directory
```
which is related to https://github.com/tweag/rules_haskell/issues/2070

File: haskell/private/cc_wrapper.py.tpl [hunk 1]
```diff
@@ -570,7 +570,10 @@ def link(output, libraries, rpaths, args):
         rpaths = shorten_rpaths(rpaths, libraries, output)
 
     args.extend(rpath_args(rpaths))
-    run_cc(args, exit_on_error=True)
+    # Note: `RULES_HASKELL_DOCTEST` is only set if called from doctest,
+    #       this is used to silence the linker output to not interfere with the output
+    #       from GHCi
+    run_cc(args, capture_output="RULES_HASKELL_DOCTEST" in os.environ, exit_on_error=True)
```
Comment (aherrmann) [inline haskell/private/cc_wrapper.py.tpl:576]: Should it maybe be called `RULES_HASKELL_SILENCE_LINKER` or something like that?

**Review by aherrmann — APPROVED** [review]
Thank you! Looks good!

File: haskell/private/cc_wrapper.py.tpl (hunk 1 above)
Comment (avdv) [reply haskell/private/cc_wrapper.py.tpl:576]: OK, :+1: I'll change that.

**Pull request merged**